client/mysqldump.c

```diff
@@ -5063,6 +5063,12 @@ static int dump_selected_tables(char *db, char **table_names, int tables)
     }
   }
 
+  /* obtain dump of routines (procs/functions) */
```

Comment:
Best to add an extra sentence to the comment above to explain that it must come before views. Ex:

```
/*
  Obtain dump of routines (procs/functions).
  This is done before dumping views as views can make use of routines.
*/
```